website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,510 @@
+                return 'selected';
+            }
+            else {
+                return 'not-selected';
+            }
+        };
+
+        // Colors asc/desc buttons on XS screen
+        self.colorSortButtonsXS = function (sort) {
+            if (self.preSelectDirection() === sort) {
+                return 'selected';
+            }
+            else {
+                return 'not-selected';
+            }
+        };
+
+        // Filtering on title
+        self.titleMatch = function (node) {
+            return (node.attributes.title.toUpperCase().indexOf(self.filter().toUpperCase()) !== -1);
+        };
+
+        // Filtering on contrib
+        self.contributorMatch = function (node) {
+            var contributors = self.contributorMapping[node.id];
+            for (var c = 0; c < contributors.length; c++) {
+                if (contributors[c].toUpperCase().indexOf(self.filter().toUpperCase()) !== -1){
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        // Filtering on tag
+        self.tagMatch = function (node) {
+            var tags = node.attributes.tags;
+            for (var t = 0; t < tags.length; t++){
+                if (tags[t].toUpperCase().indexOf(self.filter().toUpperCase()) !== -1) {
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        // Filters nodes
+        self.filterNodes = function (){
+            for (var n = 0;  n <= self.nodes().length - 1;  n++) {
+                var node = self.nodes()[n];
+                if (self.titleMatch(node) || self.contributorMatch(node) || self.tagMatch(node)) {
+                    self.eligibleNodes().push(n);
+                }
+            }
+        };
+
+        self.quickSearch = function () {
+            self.eligibleNodes([]);
+            // if backspace completely, previous nodes with prior sorting/count will be displayed
+            if (self.filter() === '') {
+                self.populateEligibleNodes(0, self.nodes().length - 1);
+            }
+            else {
+                self.filterNodes();
+            }
+        };
+
+        // Onclick, directs user to project page
+        self.nodeDirect = function(node) {
+            location.href = '/'+ node.id;
+        };
+
+    },
+    view : function(ctrl) {
+        function loadMoreButton() {
+            if (ctrl.pendingNodes()){
+                return m('button', {'class': 'col-sm-12 text-muted', onclick: function() {
+                        ctrl.loadUpToTen();
+                }},
+                m('i', {'class': 'fa fa-caret-down load-nodes'}));
+            }
+        }
+
+        function sortAlphaAsc() {
+            if (ctrl.loadingComplete()) {
+                return m('button', {id: 'alphaAsc', 'class': ctrl.colorSortButtons('alphaAsc'), onclick: function() {
+                    ctrl.sortBySortState(ctrl.sortState('alphaAsc'));
```

Comment:
This made more sense when instead of having a master node list, I had multiple lists of nodes that were displayed on the screen, nodes pending in the background, and nodes that didn't match the search query and not eligible to be displayed.  Since I was shifting nodes between lists, I was having to sort them before displaying them...so there was sorting happening even if a user didn't press a sort button. It was sorting by whatever the current sort state was.

But there is still value in tracking what the sort state is, and sorting by that sort state on the xs screen that has different controls than the full screen.